src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -115,12 +217,20 @@ private void ProcessSession(ProfilerSession session)
         {
             if (session.TryEnterPolling())
             {
-                Task.Factory.StartNew(() => 
+                Task.Factory.StartNew(() =>
                 {
                     var events = PollSession(session);
                     if (events.Count > 0)
                     {
-                        SendEventsToListeners(session.SessionId, events);
+                        // notify all viewers for the polled session
+                        List<string> viewerIDs = this.sessionViewers[session.XEventSession.ID];
+                        foreach(string vID in viewerIDs)
```

Comment:
please choose a different name for the loop variable, such as `viewerId`.  Please review the MS doc link and associated code convention pages for .Net Framework recommended naming guidelines.   Here's another good starting point https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions.

Also please add a ' ' between the 'h' and '(' in the foreach.